spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/RdbmsTaskDefinitionRepository.java

```diff
@@ -0,0 +1,297 @@
+		Assert.hasText(taskName, "taskName must not be empty nor null");
+		try {
+			return jdbcTemplate.queryForObject(prefixSql(FIND_TASK_DEFINITION_BY_NAME),
+					new TaskDefinitionRowMapper(),
+					taskName);
+		}
+		catch (EmptyResultDataAccessException e) {
+			return null;
+		}
+	}
+
+	@Override
+	public boolean exists(String taskName) {
+		Assert.hasText(taskName, "taskName must not be empty nor null");
+		boolean result;
+		try {
+			result = (jdbcTemplate.queryForObject( prefixSql(TASK_DEFINITION_COUNT_BY_NAME),
+					new Object[]{taskName}, Long.class) > 0) ? true : false;
+		}
+		catch (EmptyResultDataAccessException e) {
+			result = false;
+		}
+		return result;
+	}
+
+	@Override
+	public Iterable<TaskDefinition> findAll() {
+		return Collections.unmodifiableCollection(
+				jdbcTemplate.query(prefixSql(FIND_ALL_DEFINITIONS), new TaskDefinitionRowMapper()));
+	}
+
+	@Override
+	public Iterable<TaskDefinition> findAll(Iterable<String> taskNames) {
+		Assert.notNull(taskNames, "taskNames must not be null");
+		List<String> listOfNames = new ArrayList<String>();
+		for (String name : taskNames) {
+			listOfNames.add(name);
+		}
+
+		MapSqlParameterSource namedParameters = new MapSqlParameterSource();
+		namedParameters.addValue(LIST_OF_NAMES, listOfNames);
+
+		return Collections.unmodifiableCollection(
+				namedParameterJdbcTemplate.query(prefixSql(FIND_DEFINITIONS_WHERE_IN_CLAUSE),
+				namedParameters, new TaskDefinitionRowMapper()));
+	}
+
+	@Override
+	public long count() {
+		try {
+			return jdbcTemplate.queryForObject(
+					prefixSql(TASK_DEFINITION_COUNT), new Object[]{}, Long.class);
+		}
+		catch (EmptyResultDataAccessException e) {
+			return 0;
+		}
+	}
+
+	@Override
+	public void delete(String taskName) {
+		Assert.hasText(taskName, "taskName must not be empty nor null");
+		jdbcTemplate.update(prefixSql(DELETE_FROM_TASK_DEFINITION_BY_NAME), taskName);
+	}
+
+	@Override
+	public void delete(TaskDefinition definition) {
+		Assert.notNull(definition, "definition must not null");
+		delete(definition.getName());
+	}
+
+	@Override
+	public void delete(Iterable<? extends TaskDefinition> definitions) {
+		Assert.notNull(definitions, "definitions must not null");
+		for (TaskDefinition definition : definitions) {
+			delete(definition);
+		}
+	}
+
+	@Override
+	public void deleteAll() {
+		jdbcTemplate.update(prefixSql(DELETE_FROM_TASK_DEFINITION));
+	}
+
+	public void setTablePrefix(String tablePrefix) {
+		Assert.notNull(tablePrefix, "tablePrefix must not be null");
+		this.tablePrefix = tablePrefix;
+	}
+
+	private String prefixSql(String base) {
+		return StringUtils.replace(base, "%PREFIX%", tablePrefix);
+	}
+
+	private Page<TaskDefinition> queryForPageableResults(Pageable pageable,
+			String selectClause,
+			String fromClause,
+			String whereClause,
+			Object[] queryParam,
+			long totalCount) {
+		SqlPagingQueryProviderFactoryBean factoryBean = new SqlPagingQueryProviderFactoryBean();
```

Comment:
Creating one of these for every call may be a bit heavy since it needs to do the work of determining the type of database it's talking to each time.  I'm not saying we need to do something now, but a note here saying that we may want to improve the performance of this would be a good idea.